src/slices/posts.slice.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type Posts = {
+  loaded: boolean;
+  hasError: boolean;
+  items: Post[];
+};
+
+const initialState: Posts = {
+  loaded: false,
+  hasError: false,
+  items: [],
+};
+
+export const fetchPosts = createAsyncThunk<Post[], number>(
+  'posts/fetchPosts',
+  async (userId: number) => {
+    const posts = await getUserPosts(userId);
+
+    return posts;
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    resetPost: state => {
+      state.items = [];
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(fetchPosts.pending, state => {
+      state.loaded = true;
+      state.hasError = false;
```

Comment:
There is a logical error here. When the fetchPosts action is pending, the 'loaded' state should be set to false, indicating that the data is still loading. Setting it to true here is misleading.